UT4MasterServer/Services/AccountService.cs

```diff
@@ -18,12 +20,23 @@ public AccountService(DatabaseContext dbContext, IOptions<ApplicationSettings> s
 		allowPasswordGrant = settings.Value.AllowPasswordGrantType;
 	}
 
-	public async Task CreateAccountAsync(string username, string password)
+	public async Task CreateAccountAsync(string username, string email, string password)
 	{
-		var newAccount = new Account();
-		newAccount.ID = EpicID.GenerateNew();
-		newAccount.Username = username;
-		newAccount.Password = GetPasswordHash(newAccount.ID, password);
+		Regex regex = new Regex(@"^([\w\.\-]+)@([\w\-]+)((\.(\w){2,3})+)$");
+		Match match = regex.Match(email);
+
+		if (!accountCollection.Find(email).Equals(true)||!match.Success)
+		{
+			await CreateAccountAsync(username, email, password);
+		}
+		else
+		{
+			var newAccount = new Account();
```

Comment:
`newAccount` is not available for `await accountCollection.InsertOneAsync(newAccount);`